modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/IgniteCacheDatabaseSharedManager.java

```diff
@@ -1069,6 +1069,10 @@ public void startMemoryRestore(GridKernalContext kctx, TimeBag startTimer) throw
 
             cctx.wal(true).startAutoReleaseSegments();
             cctx.wal(true).resumeLogging(ptr);
+
+            // This callback is required for CdcManager initialization.
+            if (cctx.cdc() != null && cctx.cdc().enabled())
```

Comment:
Looks like cdc() can't return null. Do we really need null check?

Reply:
cdc() is null when cdc isn't enabled at all.